### PR TITLE
Quality of life improvements.

### PR DESCRIPTION
**Pull request by kvg**
- Modified filter command to be compatible with piped input and output.
- Specifying the model name to commands after annotate is now optional; commands will fetch the model name from the BAM header.
- Added -l <min_length> and -L <max_length> arguments to annotate to restrict read processing accordingly.
- The segment command will now remove reads without CBC/UMIs annotated unless instructed to behave otherwise.

### CODE REVIEW

File: src/longbow/extract/command.py [hunk 1]
```diff
@@ -166,11 +129,53 @@ def main(pbi, out_file, force, base_padding, leading_adapter, trailing_adapter,
             total=read_count
             ) as pbar:
 
+        # Get our model:
+        if model is None:
+            model = bam_utils.get_model_name_from_bam_header(bam_file.header)
```
Comment (jonn-smith) [inline src/longbow/extract/command.py:134]: It's possible that the user has specified a model from a JSON file.  In this case, the JSON of the model itself will be written to the bam header.

Can you add a check for this case and have it create the model from that JSON?

File: src/longbow/filter/command.py [hunk 2]
```diff
@@ -102,12 +94,29 @@ def main(pbi, out_prefix, model, force, input_bam):
             total=read_count
             ) as pbar:
 
+        # Get our model:
+        if model is None:
+            model = bam_utils.get_model_name_from_bam_header(bam_file.header)
+            lb_model = LibraryModel.build_pre_configured_model(model)
```
Comment (jonn-smith) [inline src/longbow/filter/command.py:100]: same comment about reading from the json in the BAM header.

**Review by jonn-smith — CHANGES_REQUESTED** [review]
Looks good.  I have a couple of minor change requests.

File: src/longbow/extract/command.py (hunk 1 above)
Comment (kvg) [reply src/longbow/extract/command.py:134]: Done - for every command except `annotate`, we now load the model directly from the BAM header by default.

File: src/longbow/filter/command.py (hunk 2 above)
Comment (kvg) [reply src/longbow/filter/command.py:100]: Done

**Pull request merged**